### PR TITLE
Add initial material

### PR DESCRIPTION
**Pull request by cintyaAguirre**
Please review Dockerfile, Jenkinsfile and Vagrantfiles corresponding to chapters 8, 9 and 10 for the book "Learning Continuous Integration with Jenkins"

### CODE REVIEW

File: chapter-7-continuous-integration/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    apt-get -q clean -y && rm -rf /var/lib/apt/lists/* && rm -f /var/cache/apt/*.bin &&\
+    sed -i 's|session    required     pam_loginuid.so|session    optional     pam_loginuid.so|g' /etc/pam.d/sshd &&\
+    mkdir -p /var/run/sshd
+
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+
+# Install GIT (latest edition)
+RUN apt-get -q update
```
Comment (c3s4r) [inline chapter-7-continuous-integration/Dockerfile]: It's not good to have multiple RUN commands, specially if they add files to the filesystem. Every line will generate a docker layer. Even if you remove the files later, they will increase the size of the docker image. Also, you need to add something like `apt-get -q clean -y && rm -rf /var/lib/apt/lists/* && rm -f /var/cache/apt/*.bin` at the end, to make sure that the files added by `apt-get update` are removed and don't add unnecesary size to the docker image. See: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers and https://dev.to/shriharshmishra/today-i-learned-a-simple-trick-to-slim-down-a-docker-image-ed4

File: chapter-7-continuous-integration/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+
+# Install JDK 8 (latest edition)
+RUN apt-get -q update &&\
+    DEBIAN_FRONTEND="noninteractive" apt-get -q install -y -o Dpkg::Options::="--force-confnew" --no-install-recommends software-properties-common &&\
+    add-apt-repository -y ppa:openjdk-r/ppa &&\
+    apt-get -q update &&\
+    DEBIAN_FRONTEND="noninteractive" apt-get -q install -y -o Dpkg::Options::="--force-confnew" --no-install-recommends openjdk-8-jre-headless &&\
+    apt-get -q clean -y && rm -rf /var/lib/apt/lists/* && rm -f /var/cache/apt/*.bin
+RUN apt-get -q update
```
Comment (c3s4r) [inline chapter-7-continuous-integration/Dockerfile]: Idem

File: chapter-7-continuous-integration/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+RUN apt-get -q upgrade -y
+RUN apt-get -q install -y openjdk-8-jdk
+
+# Install Maven (latest edition)
+RUN apt-get -q update
```
Comment (c3s4r) [inline chapter-7-continuous-integration/Dockerfile]: again

File: chapter-9-continuous-deployment/.vagrant/machines/node1/virtualbox/action_provision [hunk 4]
```diff
@@ -0,0 +1 @@
+1.5:31a4d843-ff5c-4d82-9b75-4c6e3932ec4c
```
Comment (c3s4r) [inline chapter-9-continuous-deployment/.vagrant/machines/node1/virtualbox/action_provision]: Shouldn't we exclude these files? Aren't they auto-generated?

File: chapter-9-continuous-deployment/.vagrant/machines/node1/virtualbox/private_key [hunk 5]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (c3s4r) [inline chapter-9-continuous-deployment/.vagrant/machines/node1/virtualbox/private_key]: We should never upload a private key to a repo. Why did you add it?

File: chapter-9-continuous-deployment/.vagrant/machines/node1/virtualbox/vagrant_cwd [hunk 6]
```diff
@@ -0,0 +1 @@
+/home/chin-pompey/Documents/tools-examples/learning-ci-jenkins-exercises-material/chapter-9-continuous-deployment
```
Comment (c3s4r) [inline chapter-9-continuous-deployment/.vagrant/machines/node1/virtualbox/vagrant_cwd]: It's an absolute path that won't work for other users...

**Review by c3s4r — CHANGES_REQUESTED** [review]
please resolve the comments...

File: chapter-7-continuous-integration/Dockerfile (hunk 1 above)
Comment (cintyaAguirre) [reply chapter-7-continuous-integration/Dockerfile]: Thanks for the reviews! I have applied those today. Could you review again please?

File: chapter-7-continuous-integration/Dockerfile (hunk 2 above)
Comment (cintyaAguirre) [reply chapter-7-continuous-integration/Dockerfile]: Thanks for the reviews! I have applied those today. Could you review again please?

File: chapter-7-continuous-integration/Dockerfile (hunk 3 above)
Comment (cintyaAguirre) [reply chapter-7-continuous-integration/Dockerfile]: Thanks for the reviews! I have applied those today. Could you review again please?

File: chapter-9-continuous-deployment/.vagrant/machines/node1/virtualbox/action_provision (hunk 4 above)
Comment (cintyaAguirre) [reply chapter-9-continuous-deployment/.vagrant/machines/node1/virtualbox/action_provision]: I have created a .gitignore file to exclude those files.

File: chapter-9-continuous-deployment/.vagrant/machines/node1/virtualbox/private_key (hunk 5 above)
Comment (cintyaAguirre) [reply chapter-9-continuous-deployment/.vagrant/machines/node1/virtualbox/private_key]: I have created a .gitignore file to exclude those files.

File: chapter-9-continuous-deployment/.vagrant/machines/node1/virtualbox/vagrant_cwd (hunk 6 above)
Comment (cintyaAguirre) [reply chapter-9-continuous-deployment/.vagrant/machines/node1/virtualbox/vagrant_cwd]: I have created a .gitignore file to exclude those files.

**Review by c3s4r — APPROVED** [review]
